adapter/src/dbt/adapters/fal_experimental/teleport.py

```diff
@@ -60,28 +62,50 @@ def _build_teleport_storage_options(teleport_info: TeleportInfo) -> Dict[str, st
         raise RuntimeError(f"Teleport storage type {teleport_info.credentials.type} not supported")
     return storage_options
 
-def run_with_teleport(code: str, teleport_info: TeleportInfo, locations: DataLocation) -> str:
+def run_with_teleport(code: str, teleport_info: TeleportInfo, locations: DataLocation, config: RuntimeConfig) -> str:
     # main symbol is defined during dbt-fal's compilation
     # and acts as an entrypoint for us to run the model.
     main = retrieve_symbol(code, "main")
-    return main(
-        read_df=_prepare_for_teleport(_teleport_df_from_external_storage, teleport_info, locations),
-        write_df=_prepare_for_teleport(_teleport_df_to_external_storage, teleport_info, locations)
-    )
+    fal_scripts_path = str(get_fal_scripts_path(config))
+    with extra_path(fal_scripts_path):
```

Comment:
I think `main = retrieve_symbol(code, "main")` has to be inside the `with`. If someone has a model like

```
from utils import my_thing

def model(dbt, fal):
  my_thing
  # ...
```